xblock/fields.py

```diff
@@ -828,31 +816,25 @@ def from_json(self, value):
         """
         Parse the date from an ISO-formatted date string, or None.
         """
-        if isinstance(value, basestring):
+        if value is None:
+            return None
 
+        if isinstance(value, basestring):
             # Parser interprets empty string as now by default
             if value == "":
                 return None
-
             try:
-                parsed_date = dateutil.parser.parse(value)
+                value = dateutil.parser.parse(value)
             except (TypeError, ValueError):
                 raise ValueError("Could not parse {} as a date".format(value))
 
-            if parsed_date.tzinfo is not None:  # pylint: disable=maybe-no-member
-                parsed_date.astimezone(pytz.utc)  # pylint: disable=maybe-no-member
-            else:
-                parsed_date = parsed_date.replace(tzinfo=pytz.utc)  # pylint: disable=maybe-no-member
-
-            return parsed_date
-
-        if value is None:
-            return None
-
-        if isinstance(value, datetime.datetime):
-            return value
+        if not isinstance(value, datetime.datetime):
+            raise TypeError("Value should be loaded from a string, not {}".format(type(value)))
```

Comment:
@e-kolpakov The method names are rather confusing.  The `from_json()` and `to_json()` methods don't actually convert from and to JSON – they just convert from and to something that is JSON-serializable, which means they can simply be the identity function in many cases.  The actual (de)serialisation is performed in `from_string()` and `to_string()`.

The only reason this `from_json()` method has to support strings is for compatibility with the legacy exports containing the spurious double quotes – otherwise `datetime.datetime` objects and `None` would be enough.

I chose some middle ground in cleaning up this API – I slightly changed the semantics of some functions where it could be done in a backwards-compatible manner, but left some inconsistencies in the code that can't be cleaned up without breaking backwards compatibility.

I will change this message to be more accurate.